googlemock/include/gmock/gmock-spec-builders.h

```diff
@@ -1471,7 +1471,14 @@ class FunctionMockerBase : public UntypedFunctionMockerBase {
   virtual ~FunctionMockerBase()
         GTEST_LOCK_EXCLUDED_(g_gmock_mutex) {
     MutexLock l(&g_gmock_mutex);
-    VerifyAndClearExpectationsLocked();
+   
+#if __cplusplus < 201703L
+    const bool inside_uncaught_exception = std::uncaught_exception();
+#else
+    const bool inside_uncaught_exception = std::uncaught_exceptions() > 0;
+#endif
+    if(!inside_uncaught_exception)
```

Comment:
Can you explain why it's okay to not `VerifyAndClearExpectationsLocked` if we're inside an unhandled exception? I'm concerned that this conditional evaluation will make an invariant more complex.